mldsa/fips202/fips202.c

```diff
@@ -368,10 +368,19 @@ __contract__(
  *
  * Arguments:   - uint64_t *s: pointer to Keccak state
  **************************************************/
-static void keccak_init(uint64_t s[25])
+static void keccak_init(uint64_t s[MLD_KECCAK_LANES])
+__contract__(
+  requires(memory_no_alias(s, sizeof(uint64_t) * MLD_KECCAK_LANES))
+  assigns(memory_slice(s, sizeof(uint64_t) * MLD_KECCAK_LANES))
+  ensures(forall(k, 0, MLD_KECCAK_LANES, s[k] == 0))
```

Comment:
This post-condition should not be needed by any callee. Please move it to an assertion at the end of the function.

Reply:
Thanks.